share/p2p/peers/manager.go

```diff
@@ -63,10 +71,14 @@ type Manager struct {
 	// hashes that are not in the chain
 	blacklistedHashes map[string]bool
 
+	// peers that have previously been removed
+	removedPeers map[peer.ID]bool
+
```

Comment:
Is it possible to get rid of this map and use full nodes pool for tracking connected peers?

Reply:
you mean by giving them a different status?

Reply:
You seem to be using this map not only for metrics:
```
func (m *Manager) isRemovedPeer(peerID peer.ID) bool {
	m.lock.Lock()
	defer m.lock.Unlock()
	return m.removedPeers[peerID]
}
```
I wonder if it is necessary to have it, since it brings a lot of junk code here an there  to keep it up to date.